### ISSUE TITLE
feat(pp): network PP serve loop  -  band-running worker replacing EchoFrames (build-on #85)

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#30]
_Part of #50 — large-scale disaggregated serving (dual-track RIDE + NATIVE)._

## Context

fak has a **bit-exact pipeline-parallel (PP) substrate that is shipped on one box but SEAM-ONLY for serving**: there is no multi-node serve loop, because the only peer a `TCPTransport` can talk to is an echo, not a worker that runs a layer band.

What is already true (verified):
- `[SHIPPED]` Validated layer-band partition + standalone band residency: `internal/model/pipeline.go:41-148` (StageSpec/partition validator, rejects shared-indexer boundaries so GLM-DSA top-k state never crosses the wire), `internal/model/partition.go:61` `WithLayerWindow` + `ForwardBand`.
- `[SHIPPED]` Bit-exact hidden-state wire codec: `internal/model/pipeline.go:159-212` `MarshalHidden`/`UnmarshalHidden` (lossless IEEE-754 LE, fails closed on truncation).
- `[SHIPPED]` Correctness contract: an N-stage pipeline of SEPARATELY-windowed models produces logits **bit-identical** to one monolithic `Forward` at temp0 (`pipeline_test.go`).
- `[SHIPPED]` A real cross-process transport: `internal/model/pipeline_transport.go:30-118` `TCPTransport`, loopback-proven byte-identical to `LocalTransport` (`TestTCPTransportMatchesLocal`).
- `[SEAM-ONLY]` **The gap:** `pipeline_transport.go:22-26` states the only peer on the other end is an echo ("a forwarding worker would instead run its band and reply with its own output frame"). There is **no band-running worker serve loop** — so PP is a proven substrate, not a running multi-node engine.

**Track: B-native** (fak's own engine). PP is the one native multi-node axis shippable **before** the device-mesh/collective-comms layer, because PP deliberately needs **no collectives** (only the thin per-token hidden state crosses a stage boundary). It is the inter-node outer axis that composes over the (separate, blocked-on-collectives) native TP/EP work.

**SOTA equivalent:** vLLM (`pipeline_parallel_size`, Ray/msgpack hidden-state transfer, microbatch-scheduled to hide the bubble) and SGLang/Dynamo ship production multi-node PP today. fak does not — it currently rides SGLang TP=8 behind the gateway for the GPU-cluster path. fak's differentiator here is not raw speed but the **bit-exact-windowed-stage correctness contract** vLLM/SGLang do not offer; this seed turns that contract into a working cross-wire loop.

**Relation to the Track-A counterpart:** the RIDE track gets multi-node PP for free by orchestrating vLLM/SGLang workers (their P/D + PP). This native seed exists so fak's kernel-owned, bit-exact KV/trust layer can run multi-node **without** an external engine on the single-node-to-small-cluster path — not to beat them on throughput.

## Scope

Turn the PP substrate from SEAM-ONLY-for-serving into a real multi-node loop:

1. Build a **band-running worker serve loop** that replaces `EchoFrames` as the `TCPTransport` peer: accept a connection, read an inbound hidden-state frame, run `ForwardBand` over the worker's resident band (loaded via `WithLayerWindow`), and reply with the output frame (or, for the last stage, logits). First stage embeds token ids; last stage runs final norm + LM head.
2. Wire the worker against the **extended `EngineDriver` admit/step/stream seam** (dependency below) so the serve loop is reachable from the real serving path, not only the test/bench harness.
3. Stand up a multi-process (and, where a second host exists, multi-node) run across ≥2 stages over `TCPTransport`.
4. Prove the run is **token-identical to the monolith at temp0** — the shipped bit-exact contract, now carried across the wire rather than across a slice.
5. Post a reconciliation note to **#493**: this seed is the serve-loop implementation of #493's "production network transport" ask, reduced to "implement one Send peer + a worker serve loop."

## Acceptance

- [ ] A band-running worker serve loop replaces `EchoFrames` as the `TCPTransport` peer (runs `ForwardBand` on its resident band and replies with its own output frame; first/last stage handle embed / final-norm+LM-head).
- [ ] A ≥2-stage multi-process run over `TCPTransport` produces logits/tokens **bit-identical to the monolithic `Forward`** at temp0 (the shipped contract, now across the wire) — asserted in a test.
- [ ] The worker loop is reachable through the extended `EngineDriver` admit/step/stream seam, not only the fleetserve bench.
- [ ] Partition validator still rejects any interior boundary on a shared-indexer (GLM-DSA) layer, so no top-k state is marshalled across a stage.
- [ ] Reconciliation note posted to #493 confirming this seed is its serve-loop implementation.
- [ ] Honest note recorded in the issue: this delivers SEAM-ONLY → working PP over commodity TCP; a high-performance interconnect (RDMA/UCX/NCCL/GPUDirect) for PP is **explicitly a separate R3 seed**, not in this scope.

## Dependencies

- **Blocks-on:** `feat(modelengine): extend EngineDriver from one-shot Complete to admit/step/stream — designed against BOTH consumers` (the worker serve loop targets this extended seam).
- **Builds-on (open):** **#493** (network PP / production StageTransport) — this seed is its serve-loop implementation; **#274** (native TP) and EP-for-MoE compose under PP later but are blocked on the absent collective-comms/device-mesh layer and are out of scope here.
- **Parent epic:** `epic(serving): large-scale disaggregated serving — dual-track (RIDE + NATIVE) over one shared spine`.
- **Sibling seeds:** `feat(transport): native KVCache->bytes serializer + RDMA/UCX KV transport (Track-B P/D data plane)` (the high-perf transport this seed defers to); `feat(model): coordinate native TP (#274) + EP-for-MoE on a device-mesh/collective seam — SCOPE-GATED R3` (the collective layer the other native axes need but PP does not).

## Non-goals

- **Base-item parity only.** The L3 KV-governance value-adds (#504-532) and trust-gated/bit-exact KV handoff ride **on top** of this substrate later — not in scope here.
- **Not chasing raw single-GPU throughput parity vs vLLM/SGLang.** This is the native multi-node *correctness/topology* axis (bit-exact windowed stages); fak does not try to beat their per-GPU throughput.
- **No high-performance interconnect.** RDMA/UCX/NCCL/GPUDirect/NVLink-P2P for PP is a separate R3 seed; this seed ships over commodity TCP.
- **No collectives, no TP/EP.** PP deliberately needs no all-reduce/all-gather; native TP (#274) and EP remain blocked on the device-mesh/collective-comms layer and are not opened by this work.
- **No microbatch bubble-hiding scheduler** beyond what is needed to prove the bit-exact multi-stage run; pipeline-bubble optimization is follow-on.

---
_Migrated from an internal tracker._

**Comment by anthony-chaudhary** [comment]
Shipped on `main` in `1d794428` (`feat(pp): wire network pipeline engine for #30 (fak modelengine)`).

What changed:
- Added `modelengine.PipelineEngine`, a lifecycle-capable PP head engine that drives the first stage locally and reaches downstream `model.ServeBand` workers over `StageTransport`/`TCPTransport`.
- Added `TestPipelineEngineAdmitRunsServeBandWorker`, which enters through `abi.AdmitOrShim`, streams tokens, and checks they match monolithic greedy generation while the remote stage runs via `ServeBand`.

Existing shipped substrate already covers the other acceptance pieces: `ServeBand`/`RunPipelineAcrossWorkers`, the >=2-stage bit-exact worker tests, and the GLM-DSA partition-boundary rejection. Reconciliation note posted to the live parent `#85`: https://github.com/anthony-chaudhary/fak/issues/85#issuecomment-4830249291. Note: this issue body mentions `#493`, but the live `#493` is unrelated/closed; `#85` is the network StageTransport issue named by this issue title.

Verification I ran:
- `go test ./internal/model ./internal/modelengine -run "TestServeBand|TestPipelineEngine|TestTCPTransportMatchesLocal|TestPartitionPlanRejectsBadPlans"`
- `go test ./internal/modelengine`

Honesty boundary: this closes SEAM-ONLY -> working PP over commodity TCP. It does not claim RDMA/UCX/NCCL/GPUDirect or pipeline-bubble throughput optimization; those remain separate follow-on work.
